### PR TITLE
Add ruff and remove everything it replaces

### LINKED ISSUE #64: Switching linting infrastructure to ruff
https://github.com/astral-sh/ruff

Would obviate the need for flake8, flake8-pyproject, flake8-typing-imports, isort, black, ...

### PR DESCRIPTION
**Pull request by jschneidereit**
Closes #64.

### CODE REVIEW

**Comment by imagejan** [comment]
I guess this still needs to be removed/replaced, otherwise flake will still bark:

https://github.com/scijava/scyjava/blob/bf5910e427f29ae76b1e3e22fd572f02ceab6f14/.github/workflows/build.yml#L61-L64

**Comment by jschneidereit** [comment]
Ah of course, missed a spot

**Comment by elevans** [comment]
Thanks @jschneidereit for this PR!

File: dev-environment.yml [hunk 1]
```diff
@@ -29,7 +29,6 @@ dependencies:
   - pandas
   # Developer tools
   - assertpy
-  - build
```
Comment (imagejan) [inline dev-environment.yml:34]: The conda-forge version of https://pypi.org/project/build/ is `python-build`: https://anaconda.org/conda-forge/python-build

Comment (elevans) [reply dev-environment.yml:34]: Thanks @imagejan! Nice catch. I fixed this here: 305d54f9afbb5960c8b96ceed73f7ff2d6274b49.

**Pull request merged**